test/e2e/singlecluster/metrics_test.go

```diff
@@ -39,9 +48,29 @@ const (
 )
 
 var _ = ginkgo.Describe("Metrics", func() {
-	ginkgo.It("should ensure the metrics endpoint is serving metrics", func() {
-		ginkgo.By("Creating a ClusterRoleBinding for the service account to allow access to metrics")
-		metricsReaderClusterRoleBinding := &rbacv1.ClusterRoleBinding{
+	var (
+		defaultGomegaMaxLength int
+
+		ns             *corev1.Namespace
+		resourceFlavor *v1beta1.ResourceFlavor
+
+		metricsReaderClusterRoleBinding *rbacv1.ClusterRoleBinding
+
+		curlContainerName string
+		curlPod           *corev1.Pod
+	)
+
+	ginkgo.BeforeEach(func() {
+		defaultGomegaMaxLength = gomegaformat.MaxLength
```

Comment:
Can you explain why we override this? And add a comment in code to clarify if we have a good usecase indeed.

Reply:
Surely. The length of the list of metrics exceeds default value, which causes it to fail, so for these tests it was overriden. Will add a comment explaining this now

Reply:
Have added an explanation to the `defaultGomegaMaxLength` variable declaration

Reply:
do you have an example log to show how and where it fails when the param is not updated? IIUC this is some internal gomega (testing framework) fail, but it is not clear to me at this point if it fails during execution or when asserting the metrics.

Reply:
The current iteration checks output per-metric to ensure that the labels are correct, so no need for the adjustments to the MaxLength, hence it was removed